### DISCUSSION TITLE
how do i purge my own package / plugin component classes?

### DISCUSSION THREAD

**Discussion by jdion84** [tailwindlabs/tailwindcss#11918] [Help]
say i am building my own component package that uses svelte for JS and tailwind classes for styling

say the package is called `my-tw-plugin`

so i add this plugin to my tailwind config:

```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [],
  theme: {
    extend: {},
  },
  plugins: [
    require('my-tw-plugin'), // here i've added my plugin
  ],
}
```

how would i get tailwind to only purge classes from components that are actually used in the project?

for example, let's say my package has a `BlueButton` component:

``js
return (
  <button class="bg-blue-600 text-white p-4">{children}</button>
)
```

and this component is used in a project page called `ContactUs`:

```js
return (
  <div>
    <h1>contact us</h1>
    <BlueButton>send message</BlueButton>
  </div>
);
```

now, let's also say that my package also has 500 other components, all with their own classes etc.

of course, in my final build i do not want to have the classes for all 500 components, that might be huge. i would only want the css for the `BlueButton` component in the final build, because that is the only component actually used in the project

how do i instruct tailwind to do this via my plugin? i can't seem to find anything about this in the docs or online

**Comment by wongjn** [comment]
You wouldn't instruct Tailwind to do that via your plugin. If the components are in separate files, you'd add file globs that only cover the files used in the `content` file globs, like:

```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [
    './node_modules/my-tw-plugin/blue-button.svelte',
  ],
  // …
}
```
